### PR TITLE
Misc updates in preparation for Vueifying config/postProcessing

### PR DESCRIPTION
**Pull request by sharkykh**
These changes were split from #4259 to make the PR lighter.

- Update `select-list` component
- Move `select-list` component registration to `main.mako`
- Update config API - postProcessing, metadata
- Standardize metadata examples
- Update store config actions

**Note:** The changes in this PR are **not** identical to the ones made on #4259.

### CODE REVIEW

File: medusa/server/api/v2/config.py [hunk 1]
```diff
@@ -187,11 +233,10 @@ class DataGenerator(object):
...
         """Get the available section names."""
-        return [
-            name[5:]
-            for (name, function) in inspect.getmembers(cls, predicate=inspect.isfunction)
-            if name.startswith('data_')
-        ]
+        cls.data__ = [name[5:] for (name, function) in inspect.getmembers(cls, predicate=inspect.isfunction) if
+                      name.startswith('data_')]
+        cls.cls_data__ = cls.data__
+        return cls.cls_data__
```
Comment (sharkykh) [inline medusa/server/api/v2/config.py]: @p0psicles 
Why did you make this change?

File: medusa/server/api/v2/config.py [hunk 2]
```diff
@@ -430,3 +507,16 @@ def data_statuses():
         section_data['strings'] = common.statusStrings
 
         return section_data
+
+    @staticmethod
+    def data_metadata():
+        """Metadata."""
+        section_data = NonEmptyDict()
+
+        section_data['metadataProviders'] = NonEmptyDict()
```
Comment (sharkykh) [inline medusa/server/api/v2/config.py]: @p0psicles 
Is there any reason for it being nested?
The metadata can just be the providers, right?

File: medusa/show/recommendations/recommended.py [hunk 3]
```diff
@@ -241,9 +241,10 @@ def update_recommended_series_cache_index(indexer, new_index):
     :param indexer: Indexer in the form of a string. For example: 'imdb', 'trakt', 'anidb'.
     :new_index: Iterable with series id's.
     """
-    index = recommended_series_cache.get(binary_type(indexer)) or set()
-    index.update(set(new_index))
-    recommended_series_cache.set(binary_type(indexer), index)
+    if new_index:
```
Comment (sharkykh) [inline medusa/show/recommendations/recommended.py]: @p0psicles 
I don't mind keep it in, but why did you make this change in your PR?
It doesn't seem to be related to postProcessing in any way.

File: medusa/server/api/v2/config.py (hunk 1 above)
Comment (p0psicles) [reply medusa/server/api/v2/config.py]: I didnt. I think someone else did.. And ik got in here through a rebase

File: medusa/server/api/v2/config.py (hunk 2 above)
Comment (p0psicles) [reply medusa/server/api/v2/config.py]: To leave room for other gneral metadata config. For example applying to all providers

File: medusa/show/recommendations/recommended.py (hunk 3 above)
Comment (p0psicles) [reply medusa/show/recommendations/recommended.py]: Aaah.. It was a bug with the anidb recommended shows. Just tested that. Ill remove it here.

**Comment by sharkykh** [comment]
Ready for review.
